src/depackers/vorbis.c

```diff
@@ -1303,14 +1303,8 @@ static void compute_bitreverse(int n, uint16 *rev)
 {
    int ld = ilog(n) - 1; // ilog is off-by-one from normal definitions
    int i, n8 = n >> 3;
-   for (i=0; i < n8; ++i) {
-      /* CID 128660 (#1 of 1): Bad bit shift operation (BAD_SHIFT)
-       * large_shift: right shifting by more than 31 bits has undefined
-       * behavior. The shift amount, 32 - ld + 3, is 36.
-       */
-      int s = 32-ld+3;
-      rev[i] = s > 31 ? 0 : (bit_reverse(i) >> s) << 2;
-   }
+   for (i=0; i < n8; ++i)
+      rev[i] = (bit_reverse(i) >> (32-ld+3)) << 2;
```

Comment:
1) Out of bounds shift occurs when `ld` is 3 or less, resulting in a shift of 32 or greater.
2) `ld` = `ilog(n) - 1` = log₂(`n`) or -1 (if `n`<=0).
3) Side note: cited shift value 36 would occur with `ld`=-1, presumably from an `n` of 0.
4) `compute_bitreverse` is only ever called from `init_blocksize` (line 1329). The `n` parameter to `compute_bitreverse` is the `n` parameter to `init_blocksize` and is never modified.
5) `init_blocksize` is called once each for both blocksizes in `start_decoder` (line 4195).
6) The blocksizes are calculated from stored log₂() values in the Vorbis stream, also in `start_decoder` (line 3670). These log values are bounded between 6 and 13, meaning the blocksizes will never be less than 64 or greater than 8192.
7) `ilog(64)` = `(5 + log2_4[64 >> 5])` = `(5 + log2_4[2])` = 7, so `ld` = 6 is the minimum value for `ld`.

This hack was added in this patch: https://github.com/libxmp/libxmp/commit/db5cad087d0baf36c5368bdfb9f2aa3ccf02f8b4.

The same analysis holds up for the copy of stb_vorbis that was originally patched, meaning this fix never actually did anything and was just to shut up a static analyzer false positive.